common/docker/script.go

```diff
@@ -0,0 +1,128 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"pkg.world.dev/world-cli/common/logger"
+)
+
+var (
+	// mountCache is the Docker mount command to cache the go build cache
+	mountCache = `--mount=type=cache,target="/root/.cache/go-build"`
+
+	// cardinalDockerfile is the Dockerfile for the Cardinal service
+	cardinalDockerfile string
+
+	buildKitSupport bool
+)
+
+func init() {
+	// Check if the Docker version supports BuildKit
+	if buildKitSupport = checkBuildKitSupport(); !buildKitSupport {
+		// If BuildKit is not supported, disable the go build cache mounting
+		mountCache = ""
+		logger.Info("BuildKit is not supported")
+	} else {
+		logger.Info("BuildKit is supported")
+	}
+
+	cardinalDockerfile = fmt.Sprintf(`
+	################################
```

Comment:
You can put this inside a regular Dockerfile and embed it to the binary using the go embed directive

https://gobyexample.com/embed-directive

Reply:
moved to `cardinal.Dockerfile` and embed it.